ERRORCODES.md

```diff
@@ -0,0 +1,73 @@
+| *10020* | The endUserClientId provided is not valid|
+| *10021* | Enter a valid Consent Status Enum value|
+| *10022* | Cannot update Consent Status where the Consent Owner of the channel is not set to Partner|
+| *10023* | Cannot update Consent Status where the SupportsKeywords of the channel is true|
+| *10024* | For Flow based Routing, flow ID and Fallback Queue ID are required fields|
+| *10025* | The conversation id isn't found based on channelAddressIdentifier and CUSTOM channel type.|
+| *10026* | The conversaction channel defnition for the provided *"to":"\<channelAddressIdentifier\>"* field is missing.|
+| *10027* | The provided *"to"* field isn't valid.|
+| *10028* | The provided *"from"* field isn't valid.|
+| *10029* | The provided interaction list isn't valid.|
+| *10030* | Specify either a flow or a queue to Route API|
+| *10031* | The channelAddressIdentifier provided is not active.|
+| *10032* | Specified Queue ID is invalid|
+| *10033* | Should not pass in error message when routing result success.|
+| *10034* | Error message missing when routing result false.|
+| *10035* | The messaging channel is not a type of CUSTOM.|
+| *10036* | The participant does not exist|
+| *10037* | The AppType provided is not supported|
+| *10038* | The Participant Role provided is not supported|
+| *10039* | Add participants to your conversation.|
+| *10040* | Provide a participant role.|
+| *10041* | Provide a participant subject.|
+| *10042* | Include one end user participant in the conversation.|
+| *10043* | Invalid participant role. Specify a Chatbot or EndUser role for all participants.|
+| *10044* | Include only one end user participant in the conversation.|
+| *10045* | Provide a valid message entry format.|
+| *10046* | Invalid clientTimestamp. Send messages within the allowed response time frame.|
+| *10047* | Specified Message Type is invalid|
+| *10048* | Specified Format Type is invalid|
+| *10049* | Specified Message Type and Format Type combination is invalid|
+| *10050* | Specified custom integration is invalid|
+| *10051* | The file size (in bytes) of uploaded attachment is incorrect.|
+| *10052* | The "channelAddressIdentifier" field set in post conversation request isn't valid.|
+| *10053* | The "participants" field set in post conversation request is either invalid or empty.|
+| *10054* |Invalid subject of participant in conversation request.|
+| *10055* | Conversation precondition failed. Please retry. |
+| *10056* | Conversation processing aborted. Please retry. |
+| *10057* | Looks like you don't have access to the ConversationHistory and AttachmentHistory APIs. Only Bring Your Own Channel for CCaaS partners can access them. Your Salesforce admin can help with that. |
+
+## Service Errors
+| Error Code | Description |
+|--|--|
+| *20002* | An error occured when retrieve conversation list based on endUserAddress. |
+| *20003* | Empty interaction list in request.|
```

Comment:
If this is complaining about input. Shouldn't this be a 400 error rather than 500? Is this error code used anywhere? By doing quick search I did not find any reference. if not used can we please remove it from here?

Reply:
yeah, I agree with it, it should be 400 error, due to it is an invalid param in request payload.

Reply:
@jselvam-sf What do you think this error code? it should be 400 error instead.

Reply:
We can keep it as-is for now. since we need to update the code definition in scrt2-interaction-service ApiErrorEnum too. Please file a WI, and fix it later.